calico_node/startup/startup.go

```diff
@@ -906,6 +913,84 @@ func ensureDefaultConfig(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c
 	return nil
 }
 
+type ClientV1 string
+
+func (_ ClientV1) Blah() (string, bool, error) {
+	return "v2.6.1", true, nil
+}
+
+// ensureMigrated ensures any data migration needed is done.
+func ensureMigrated(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c client.Interface) error {
+	var cv1 ClientV1
+
+	// Only run this migration if using KDD
+	if cfg.Spec.DatastoreType != apiconfig.Kubernetes {
```

Comment:
nit: it's cleaner to perform this check outside of the function. It makes the scope of this function more defined, and the calling code easier to read.